xml/art_virtualization-best-practices.xml

```diff
@@ -3241,6 +3241,162 @@ LC_ALL=C PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
       </para>
 <screen>&prompt.root;<command>echo 1 > /sys/devices/system/cpu/cpu<replaceable>X</replaceable>/online</command></screen>
     </sect2>
+
+    <sect2 xml:id="sec-vm-guest-scsi-pr">
+      <title>SCSI3 persistent reservation (S3-PR) on a multipathed device</title>
+      <para>
+        SCSI persistent reservations allow restricting access to block devices
+        to specific initiators in a shared storage setup.  When implementing
+        clustering of &vmguest;, it is a common requirement for &vmguest; to
+        send persistent reservation SCSI commands.  However, the operating
+        system restricts sending these commands to unprivileged programs
+        because incorrect usage can disrupt regular operation of the storage
+        fabric. QEMU's SCSI passthrough devices <literal>scsi-block</literal>
+        and <literal>scsi-generic</literal> support passing guest persistent
+        reservation requests to a privileged external helper program.
+        The <literal>qemu-pr-helper</literal> is that external helper, It needs
+        to start before QEMU, creates a listener socket which will accept
+        incoming connections for communication with QEMU.
```

Comment:
A step by step guide on how to set this up would be better I think rather than having the user parse the docs and try to pick up elements from different paragraphs.

Something like:

1. What is multipathed I/O (maybe a reference to SLE-doc https://github.com/SUSE/doc-sle/blob/main/xml/storage_multipath.xml ?
2. What are scsi peristent reservation on multipathed I/O and why are they important
3. step - by - step guide on how to set them up correctly (does the helper need to be started manually? By libvirt? when, how?
4. what are the pitfalls and how to avoid them.

Reply:
About 1,  Yes,  a reference is good idea.

About 2, It's out of virtualization scope, It should be explained in storage doc.

About 3,
It's just a brief introduction about scsi pr helper from qemu's angle.

No commercial users use qemu directly, Usually they use upper management tools(say libvirt) to manage their vm instances.

from libvirt's angle: I already presented an example of disk element about how to correctly set up scsi pr, and already mentioned that libvirt will handle everything else if user set the 'managed' to 'yes' then libvirt prepares and manages any resources needed.

From qemu's angle: Do you think I should present such a step by step example? 
e.g. How to manually launch a qemu-pr-helper instance then manually launch related qemu instance?

Reply:
agree on 2., but what I mean is why is multipathed I/O important _for virtualization_ , this is the part that I think fits into this document as opposed to the storage document.

About libvirt/qemu: no I think we should describe well in our step-by-step thing assuming that libvirt is in the picture. So if libvirt manages the qemu pr helper itself there is little reason to even mention it here I think, if it is nothing that the user should actually act on.

The elements are all here, I just suggest to put it into 1-2-3-4-5-6 actionable steps, and follow up with examples later, so that a busy sysadmin can just focus on following the steps, and look at examples if necessary for clarification. Thanks a lot!

Reply:
"why is multipathed I/O important for virtualization"?
The answer is in why it is important for bare metal.

Just like why is 'apic' important for virtualization? The answer is in why it is important for real SMP.

Reply:
Why the customers need scsi pr in virtualization?
Because they use this mechanism to protect their shared scsi storage in real world, then they wants it in virtualization as well.